app/spa.html

```diff
@@ -0,0 +1,265 @@
+            }
+
+            function createHashChange() {
+                createXMLRequest();
+                location.hash = 'hash_change';
+                createXMLRequest();
+                beginMutation(50);
+            }
+
+            function createHashChangeWithPushState() {
+                createXMLRequest();
+                window.history.pushState(
+                    { state: 'two' },
+                    'Page Two',
+                    '#hash_changed_pushState'
+                );
+                createXMLRequest();
+                beginMutation(50);
+            }
+            function back() {
+                createXMLRequest();
+                window.history.back();
+                createXMLRequest();
+                beginMutation(50);
+            }
+
+            function back() {
+                createXMLRequest();
+                window.history.back();
+                createXMLRequest();
+                beginMutation(50);
+            }
+
+            function forward() {
+                createXMLRequest();
+                window.history.forward();
+                createXMLRequest();
+                beginMutation(50);
+            }
+
+            function go(number) {
+                createXMLRequest();
+                window.history.go(number);
+                createXMLRequest();
+                beginMutation(50);
+            }
+
+            function beginMutation(epoch) {
+                let i = 0;
+                function helper() {
+                    domMutation();
+                    if (i++ < epoch) {
+                        console.log('MUTATING');
+                        setTimeout(helper, 10);
+                    }
+                }
+                helper();
+            }
+
+            function dispatch() {
+                cwr('dispatch');
+            }
+
+            function clearRequestResponse() {
+                document.getElementById('request_url').innerText = '';
+                document.getElementById('request_header').innerText = '';
+                document.getElementById('request_body').innerText = '';
+
+                document.getElementById('response_status').innerText = '';
+                document.getElementById('response_header').innerText = '';
+                document.getElementById('response_body').innerText = '';
+            }
+
+            function timeoutLoad() {
+                createXMLRequest();
+                window.history.pushState(
+                    { state: 'two' },
+                    'Page Two',
+                    '/page_view_two?search=foo#hash2'
+                );
+                beginMutation(200);
+            }
+
+            function createXMLRequest() {
+                var xhr = new XMLHttpRequest((async = true));
+                var url =
+                    'https://dataplane.rum.us-west-2.amazonaws.com/appmonitors/abc123/';
+                xhr.open('GET', url, true);
+                xhr.send();
```

Comment:
issue: Could we access a local resources here instead of dataplane?

Reply:
Agreed, I will replace it with the existing example endpoint we are using in other test applications.